src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_DeconstructionAssignmentOperator.cs

```diff
@@ -103,6 +106,88 @@ internal sealed partial class LocalRewriter
 
                 return _factory.Sequence(temps.ToImmutableAndFree(), effects.ToImmutableAndFree(), returnValue);
             }
+
+            void reverseAssignmentsToTargetsIfApplicable()
+            {
+                // The list of deconstruction targets is usually small. Prefer a temporary array which is stack-bound for small numbers of elements.
+                var visitedSymbols = TemporaryArray<Symbol>.Empty;
```

Comment:
>TemporaryArray<Symbol>.Empty; [](http://example.com/codeflow?start=37&length=29)

I would use pooled hashset or a SmallDictionary since we are looking for members in the array

Reply:
I'm open to changing this, but will hold off until we have decided more about how we want this optimization to work. For example, do we want to break behavior of any possibly aliasing assignments in order to get better codegen in some scenarios we care about.